pkg/provider/azure_vmss_test.go

```diff
@@ -2262,6 +2262,7 @@ func TestEnsureVMSSInPool(t *testing.T) {
 		vmSetName             string
 		clusterIP             string
 		nodes                 []*v1.Node
+		mutate                func(compute.VirtualMachineScaleSet)
```

Comment:
I think this `mutate()` is to turn a linux vmss to a windows one, but it doesn't seem clear with just "mutate" as a variable name. Then I suggest a better one:
1. create a `buildTestWindowsVMSSWithLB()` and in this func, call `buildTestVMSSWithLB()` and mutate there
2. use a `isWindows2019` variable instead here, then in this test func L2556, create a vmss with `buildTestVMSSWithLB()` or `buildTestWindowsVMSSWithLB()`

Reply:
hm, `mutate` allows the test writer to change any property they want so that we don't have to keep adding more params to BuildTest*() funcs. this makes things more extensible. but I don't mind changing it to have explicit windows semantics.

Reply:
okay, refactored in https://github.com/kubernetes-sigs/cloud-provider-azure/pull/6316/commits/621d00871ede291ac367cbfae1b1bc5157e9eeb1